### DISCUSSION TITLE
Ubuntu: Build from GIT source

### DISCUSSION THREAD

**Discussion by fdcastel** [troglobit/inadyn#453] [Show and tell]
For reference: Compilation of steps needed to build `inadyn` on latest Ubuntu Server LTS (22.04).

1) First, this one to avoid goddam popups during package installs ([source](https://stackoverflow.com/questions/73397110/how-to-stop-ubuntu-pop-up-daemons-using-outdated-libraries-when-using-apt-to-i)):
    ```bash
    # Ubuntu 22.04 LTS: Change needrestart from interactive to automatic
    sudo sed -i "s/nrconf{restart} = 'i'/nrconf{restart} = 'a'/g" /etc/needrestart/needrestart.conf
    reboot
    ```

2) Then install the required dependencies:
    ```bash
    # Install required dependencies
    sudo apt install -y git automake autoconf 
    sudo apt install -y libtool build-essential pkg-config libconfuse-dev gnutls-dev    # Not in README.md (!)
    ```

3) Finally, make it so:
    ```bash
    # Clone and build
    cd ~
    git clone https://github.com/troglobit/inadyn.git
    cd inadyn/
    ./autogen.sh
    ./configure && make
    ```

Output binary will be in `./src/inadyn`.

**Comment by fdcastel** [comment]
P.S.: Soon I will be in dire need to cross-build the code for `MIPS` architecture (used in Ubiquiti EdgeRouters).

If anybody with experience in this subject could point me in the right direction, it would help immensely.

**Reply by troglobit** [reply]
Should be fairly straight-forward, you can have a look at how Buildroot does it, or test yourself with a crosstool-NG toolchain.  There is also the Bootlin (maintainers of Buildroot) pre-built toolchains: <https://toolchains.bootlin.com/>

Usually what you need is:
 1. The toolchain's bin directory in your $PATH (where the mips-linux-gcc and mips-linux-ld, etc. are)
 2. Build your program and any deps. with `configure --host=mips-linux-`, and any other args you need, e.g., the relative install path inside your sysroot `--prefix=/usr`,
 3. Call `make`
 4. Set `DESTDIR=/path/to/sysrooot` and call `make install`
 5. inadyn binary should now be in `/path/to/sysroot/usr/sbin/inadyn`

There are some tricks to getting configure to find the .pc file for library dependencies, but try the above first to get going.
